### PR TITLE
Devise translations

### PR DESCRIPTION
**Pull request by rafamanzo**

### CODE REVIEW

File: app/views/devise/_links.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<% ActiveSupport::Deprecation.warn "Rendering partials devise/_links.erb is deprecated" \
+  "please use devise/shared/_links.erb instead."%>
+<%= render "shared/links" %>
```
Comment (diegoamc) [inline app/views/devise/_links.erb]: Is "devise/_links.erb" rendered automatically by devise? If not, why do we need this file? Can't we just render "shared/links"

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 2]
```diff
@@ -1,5 +1,6 @@
-<p>Welcome <%= @email %>!</p>
+<% require 'devise/version' %>
+<p><%= t('.greeting', :recipient => @resource.email, :default => "Welcome #{@resource.email}!") %></p>
 
-<p>You can confirm your account email through the link below:</p>
-
-<p><%= link_to 'Confirm my account', confirmation_url(@resource, :confirmation_token => @token) %></p>
+<p><%= t('.instruction', :default => "You can confirm your account email through the link below:") %></p>
+<p><%= link_to t('.action', :default => "Confirm my account"),
+         confirmation_url(@resource, :confirmation_token => (Devise::VERSION.start_with?('3.') ? @token : @resource.confirmation_token)) %></p>
```
Comment (diegoamc) [inline app/views/devise/mailer/confirmation_instructions.html.erb]: Isn't there a better way to do this? We need to use "@resource.confirmation_token" with versions 2.* or 1.*? What will happen when version 4.* is released?

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 3]
```diff
@@ -1,8 +1,9 @@
-<p>Hello <%= @resource.email %>!</p>
+<% require 'devise/version' %>
+<p><%= t('.greeting', :recipient => @resource.email, :default => "Hello #{@resource.email}!") %></p>
 
-<p>Someone has requested a link to change your password. You can do this through the link below:</p>
+<p><%= t('.instruction', :default => "Someone has requested a link to change your password, and you can do this through the link below.") %></p>
 
-<p><%= link_to 'Change my password', edit_password_url(@resource, :reset_password_token => @token) %></p>
+<p><%= link_to t('.action', :default => "Change my password"), edit_password_url(@resource, :reset_password_token => (Devise::VERSION.start_with?('3.') ? @token : @resource.reset_password_token)) %></p>
```
Comment (diegoamc) [inline app/views/devise/mailer/reset_password_instructions.html.erb]: Same thing here.

File: app/views/devise/mailer/unlock_instructions.html.erb [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
+<p><%= t('.greeting', :recipient => @resource.email, :default => "Hello #{@resource.email}!") %></p>
 
-<p>Your account has been locked due to an excessive number of unsuccessful sign in attempts.</p>
+<p><%= t('.message', :default => "Your account has been locked due to an excessive amount of unsuccessful sign in attempts.") %></p>
 
-<p>Click the link below to unlock your account:</p>
+<p><%= t('.instruction', :default => "Click the link below to unlock your account:") %></p>
 
-<p><%= link_to 'Unlock my account', unlock_url(@resource, :unlock_token => @token) %></p>
+<p><%= link_to t('.action', :default => "Unlock my account"), unlock_url(@resource, :unlock_token => (Devise::VERSION.start_with?('3.') ? @token :@resource.unlock_token)) %></p>
```
Comment (diegoamc) [inline app/views/devise/mailer/unlock_instructions.html.erb]: Same thing here.

File: app/views/devise/confirmations/new.html.erb [hunk 5]
```diff
@@ -1,14 +1,12 @@
...
 
 <%= form_for(resource, :as => resource_name, :url => confirmation_path(resource_name), :html => { :method => :post }) do |f| %>
   <%= devise_error_messages! %>
 
   <div><%= f.label :email %><br />
-  <%= f.email_field :email, :autofocus => true %></div>
+  <%= f.email_field :email %></div>
 
-  <div><%= f.submit "Resend confirmation instructions", class: 'btn btn-default' %></div>
+  <div><%= f.submit t('.resend_confirmation_instructions', :default => 'Resend confirmation instructions') %></div>
```
Comment (diegoamc) [inline app/views/devise/confirmations/new.html.erb]: We won't use class btn btn-default anymore?

File: app/views/devise/mailer/unlock_instructions.html.erb (hunk 4 above)
Comment (diegoamc) [inline app/views/devise/mailer/unlock_instructions.html.erb]: Same thing here.

File: app/views/devise/registrations/edit.html.erb [hunk 6]
```diff
@@ -1,8 +1,7 @@
 <div class="page-header">
-  <h1>Edit <%= resource_name.to_s.humanize %></h1>
-</div>
+  <h1><%= t('.title', :resource => resource_class.model_name.human , :default => "Edit #{resource_name.to_s.humanize}") %></h1>
 
-<%= form_for(resource, :as => resource_name, :url => registration_path(resource_name), :html => { :method => :put, :class => "form-inline" }) do |f| %>
+<%= form_for(resource, :as => resource_name, :url => registration_path(resource_name), :html => { :method => :put }) do |f| %>
```
Comment (diegoamc) [inline app/views/devise/registrations/edit.html.erb]: Why remove the option form-inline?

File: app/views/devise/registrations/edit.html.erb [hunk 7]
```diff
@@ -54,32 +53,32 @@
...
+      </div>
+    </div>
   </div>
 <% end %>
 
-<hr />
+<h2><%= t('.cancel_my_account', :default => 'Cancel my account') %></h2>
 
-<h2>Cancel my account</h2>
+<p><%= t('.unhappy', :default => 'Unhappy') %>? <%= link_to t('.cancel_my_account', :default => "Cancel my account"), registration_path(resource_name), :data => { :confirm => t('.are_you_sure', :default => "Are you sure?") }, :method => :delete %>.</p>
```
Comment (diegoamc) [inline app/views/devise/registrations/edit.html.erb]: We won't use the btn-danger style anymore?

File: app/views/devise/registrations/new.html.erb [hunk 8]
```diff
@@ -1,8 +1,8 @@
 <div class="page-header">
-  <h1>Sign Up</h1>
+  <h1><%= t('.sign_up', :default => "Sign up") %></h1>
 </div>
 
-<%= form_for(resource, :as => resource_name, :url => registration_path(resource_name),:html => { :class => "form-inline" }) do |f| %>
+<%= form_for(resource, :as => resource_name, :url => registration_path(resource_name)) do |f| %>
```
Comment (diegoamc) [inline app/views/devise/registrations/new.html.erb]: Option form-inline remvoed here either.

File: app/views/devise/sessions/new.html.erb [hunk 9]
```diff
@@ -1,63 +1,51 @@
 <div class="page-header">
-  <h1>Login into Mezuro</h1>
+  <h1><%= t('.sign_in', :default => "Sign in") %></h1>
 </div>
 
-<%= form_for(resource, :as => resource_name, :url => session_path(resource_name),:html => { :class => "form-inline" }) do |f| %>
+<%= form_for(resource, :as => resource_name, :url => session_path(resource_name)) do |f| %>
```
Comment (diegoamc) [inline app/views/devise/sessions/new.html.erb]: Option form-inline removed here either.

File: app/views/devise/unlocks/new.html.erb [hunk 10]
```diff
@@ -1,14 +1,12 @@
...
 
 <%= form_for(resource, :as => resource_name, :url => unlock_path(resource_name), :html => { :method => :post }) do |f| %>
   <%= devise_error_messages! %>
 
   <div><%= f.label :email %><br />
-  <%= f.email_field :email, :autofocus => true %></div>
+  <%= f.email_field :email %></div>
 
-  <div><%= f.submit "Resend unlock instructions", class: 'btn btn-primary' %></div>
+  <div><%= f.submit t('.resend_unlock_instructions', :default => "Resend unlock instructions") %></div>
```
Comment (diegoamc) [inline app/views/devise/unlocks/new.html.erb]: We won't use btn-primary anymore?

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -77,7 +77,7 @@
                   <span class="caret"></span>
                 </a>
                 <ul class="dropdown-menu" role="menu">
-                  <% I18n.available_locales.each do |locale| %>
+                  <% [:en, :pt].each do |locale| %>
```
Comment (diegoamc) [inline app/views/layouts/application.html.erb]: Why did you have to do this change?

File: app/views/devise/mailer/confirmation_instructions.html.erb (hunk 2 above)
Comment (rafamanzo) [inline app/views/devise/mailer/confirmation_instructions.html.erb]: I don't know how to answer your questions. This was auto-generated by ```devise-i18n-views```, which I suppose knows better than I what's the best in this case.

File: app/views/devise/mailer/reset_password_instructions.html.erb (hunk 3 above)
Comment (rafamanzo) [inline app/views/devise/mailer/reset_password_instructions.html.erb]: Same.

File: app/views/devise/mailer/unlock_instructions.html.erb (hunk 4 above)
Comment (rafamanzo) [inline app/views/devise/mailer/unlock_instructions.html.erb]: Same

File: app/views/devise/registrations/edit.html.erb (hunk 6 above)
Comment (rafamanzo) [inline app/views/devise/registrations/edit.html.erb]: We mieed this after the view regeneration. Let's fix it.

File: app/views/devise/registrations/edit.html.erb (hunk 7 above)
Comment (rafamanzo) [inline app/views/devise/registrations/edit.html.erb]: This was auto-generated by devise-i18n-views, which I suppose knows better than I what's the best in this case. Probably why it has changed from link to button.

I agree to restore the ```danger``` style back.

File: app/views/devise/registrations/new.html.erb (hunk 8 above)
Comment (rafamanzo) [inline app/views/devise/registrations/new.html.erb]: Let's restore here as well.

File: app/views/devise/sessions/new.html.erb (hunk 9 above)
Comment (rafamanzo) [inline app/views/devise/sessions/new.html.erb]: Let's restore here as well.

File: app/views/devise/unlocks/new.html.erb (hunk 10 above)
Comment (rafamanzo) [inline app/views/devise/unlocks/new.html.erb]: We forgot to restore this. Let's fix it.

File: app/views/layouts/application.html.erb (hunk 11 above)
Comment (rafamanzo) [inline app/views/layouts/application.html.erb]: I was changed before ```devise-i18n``` was removed from Gemfile. We need to undo this modification.

File: app/views/devise/_links.erb (hunk 1 above)
Comment (rafamanzo) [inline app/views/devise/_links.erb]: I don't know how to answer your question. This was auto-generated by devise-i18n-views, which I suppose knows better than I what's the best in this case.

File: app/views/devise/confirmations/new.html.erb (hunk 5 above)
Comment (rafamanzo) [inline app/views/devise/confirmations/new.html.erb]: We forgot this style. We need to restore it.

**Comment by rafamanzo** [comment]
Done all the fixes that I commented that were possible. Waiting for travis and a review for the new code.

**Comment by rafamanzo** [comment]
Travis failled but just the test that sometimes fails according to the machine. We may consider increase capybara timeout from 5s to 10s.

**Comment by danielkza** [comment]
Do we really need set the timeout to something relatively low? Is there any downside to making it something like 30s straight away to never have to worry about it later? Would it make tests that don't reach the timeout slower, or is Capybara/Poltergeist smart enough to find out immediately when the page is ready?

File: app/views/devise/confirmations/new.html.erb [hunk 12]
```diff
@@ -1,14 +1,12 @@
-<div class="page-header">
-  <h1>Resend confirmation instructions</h1>
-</div>
+<h2><%= t('.resend_confirmation_instructions', :default => 'Resend confirmation instructions') %></h2>
```
Comment (pedroscocco) [inline app/views/devise/confirmations/new.html.erb]: Needs class page-header

File: app/views/devise/confirmations/new.html.erb [hunk 13]
```diff
@@ -1,14 +1,12 @@
...
 
 <%= form_for(resource, :as => resource_name, :url => confirmation_path(resource_name), :html => { :method => :post }) do |f| %>
   <%= devise_error_messages! %>
 
   <div><%= f.label :email %><br />
-  <%= f.email_field :email, :autofocus => true %></div>
+  <%= f.email_field :email %></div>
```
Comment (pedroscocco) [inline app/views/devise/confirmations/new.html.erb]: autofocus = true

**Comment by rafamanzo** [comment]
I believe 30s is still reasonable. I suggested to go with 10s to try to
keep it as low as possible so the tests would fail as fast as possible.

Em qua, 15 de abr de 2015 às 16:12, Daniel Miranda <notifications@github.com>
escreveu:

File: app/views/devise/passwords/edit.html.erb [hunk 14]
```diff
@@ -1,18 +1,16 @@
-<div class="page-header">
-  <h1>Change your password</h1>
-</div>
+<h2><%= t('.change_your_password', :default => 'Change your password') %></h2>
```
Comment (pedroscocco) [inline app/views/devise/passwords/edit.html.erb]: class: page-head

File: app/views/devise/unlocks/new.html.erb [hunk 15]
```diff
@@ -1,14 +1,12 @@
...
 
   <div><%= f.label :email %><br />
-  <%= f.email_field :email, :autofocus => true %></div>
+  <%= f.email_field :email %></div>
 
-  <div><%= f.submit "Resend unlock instructions", class: 'btn btn-primary' %></div>
+  <div><%= f.submit t('.resend_unlock_instructions', :default => "Resend unlock instructions"), class: 'btn btn-primary' %></div>
 <% end %>
 
 <%= render "devise/shared/links" %>
```
Comment (pedroscocco) [inline app/views/devise/unlocks/new.html.erb]: Fix styles

Comment (rafamanzo) [inline app/views/devise/unlocks/new.html.erb]: Which style? Here the submit class was preserved.

**Comment by rafamanzo** [comment]
Done with what @pedroscocco pointed out.

File: app/views/devise/unlocks/new.html.erb (hunk 15 above)
Comment (pedroscocco) [inline app/views/devise/unlocks/new.html.erb]: Page header at the top

**Pull request merged**